HomeWork1/src/index.ts

```diff
@@ -0,0 +1,127 @@
+// // Ви маєте JS код, який потрібно перевести у TS та додати анотацію типів до примітивів. Не чіпайте обʼєкти та будь що, чого ми ще не розглядали на занятті. Замість цього ви можете зараз використовувати any, поки ми не знаємо кращого вибору. Перевіряйте код, в ньому є речі, які потрібно додати до класів, не тільки типи. Вам треба зробити рефакторинг, будьте уважні.
+
+class School {
+  private directions: string[] = [];
+
+  addDirection(direction: string): void {
+    this.directions.push(direction);
+  }
+}
+
+class Direction {
+  private levels: any[] = [];
+  private _name: string;
+
+  get name(): string {
+    return this._name;
+  }
+
+  constructor(name: string) {
+    this._name = name;
+  }
+
+  addLevel(level: any): void {
+    this.levels.push(level);
+  }
+}
+
+class Level {
+  private groups: any[] = [];
+  private _name: string;
+  private _program: any;
+
+  constructor(name: string, program: any) {
+    this._name = name;
+    this._program = program;
+  }
+
+  get name(): string {
+    return this._name;
+  }
+
+  get program(): any {
+    return this._program;
+  }
+
+  addGroup(group: any): void {
+    this.groups.push(group);
+  }
+}
+
+class Group {
+  _students: string[] = [];
+  directionName: string;
+  levelName: string;
+
+  get students(): string[] {
+    return this._students;
+  }
+
+  constructor(directionName: string, levelName: string) {
+    this.directionName = directionName;
+    this.levelName = levelName;
+  }
+
+  addStudent(student: string): void {
+    this._students.push(student);
+  }
+
+  showPerformance(): string[] {
+    const sortedStudents = this._students.toSorted(
```

Comment:
Just a friendly reminder that you can improve the predictability and safety of your code by explicitly adding type annotations for constants.